### PR TITLE
backend/fix :#38 added new apis for user sos

### PR DESCRIPTION
**Pull request by soumyajit2304**
- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description
Added a APIs for creating user sos entries.

### Additional Changes

- [x] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

Provide links to the files with corresponding changes.

You can find config files in `dhall-configs`

## Checklist

- [x] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Profile.hs [hunk 1]
```diff
@@ -63,3 +90,39 @@ updatePerson personId req = do
...
+  return $
+    SosRes
+      { sosId = sosDetails.id
+      }
+
+updateSosDetails :: (EsqDBFlow m r, EncFlow m r) => Id Person.Person -> SosFeedbackReq -> m APISuccess.APISuccess
+updateSosDetails personId req = do
+  sosDetails <-
+    QSos.findById (req.sosId)
+      >>= fromMaybeM SosIdDoesNotExist
```
Comment (Dwibedi-10) [inline Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Profile.hs:107]: You can send sosId also while throwing this error

File: Backend/app/rider-platform/rider-app/Main/src/Tools/Error.hs [hunk 2]
```diff
@@ -50,3 +50,17 @@ instance IsHTTPError EstimateError where
   toHttpCode _ = E400
 
 instance IsAPIError EstimateError
+
+data SosError
```
Comment (Dwibedi-10) [inline Backend/app/rider-platform/rider-app/Main/src/Tools/Error.hs:54]: Create this ErrorType in shared kernel

File: Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Profile.hs (hunk 1 above)
Comment (soumyajit2304) [reply Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Profile.hs]: > You can send sosId also while throwing this error

Done

**Review by hkmangla — CHANGES_REQUESTED** [review]

File: Backend/stack.yaml [hunk 3]
```diff
@@ -81,7 +81,7 @@ extra-deps:
       - app/mock-registry
   # -  ../../shared-kernel/lib/mobility-core
   - git: https://github.com/nammayatri/shared-kernel.git
-    commit: 0655c0de03b86fa11d30e4212af4f11d248928c1
+    commit: 97beaeada83a9cbddced0f34e98846dfd1579cb9
```
Comment (hkmangla) [inline Backend/stack.yaml:84]: This is incorrect value. Update it with latest hash of shared-kernel d09348f46006c60ace8c613b75a66ba33fcf9a72

**Review by hkmangla — CHANGES_REQUESTED** [review]

**Review by hkmangla — APPROVED** [review]

**Pull request merged**